### PR TITLE
Append `solve` to http solver url path

### PR DESCRIPTION
**Pull request by gio256**
Previously, http solver queries used [`reqwest::Url::set_path()`](https://docs.rs/reqwest/latest/reqwest/struct.Url.html#method.set_path), which replaced any path in the solver url with `/solve`. This change replaces this with [`reqwest::Url::join()`](https://docs.rs/reqwest/latest/reqwest/struct.Url.html#method.join), which instead appends `solve` to the existing url path.

This allows external solver urls to include a path. Previously, requests for a solver at `https://solver.com/barn/` and a solver at `https://solver.com/prod/` would both get sent to `https://solver.com/solve`, because the url path was dropped. With this change, the requests will go to `https://solver.com/barn/solve` and `https://solver.com/prod/solve`, respectively.

Note that the url path must include a trailing slash, or `reqwest::Url::join()` will interpret the last element of the path as a filename and replace it. For example, a solver at `https://solver.com/foo/bar` will result in requests sent to `https://solver.com/foo/solve` because there is no trailing slash.

### Test Plan

There are currently no tests, though I am happy to add some if there is a good way to test this change. This would be a breaking change if there are currently any external solvers with a url path relying on that path being dropped.

### Release notes

None.

### CODE REVIEW

**Comment by gio256** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by josojo — APPROVED** [review]
Thanks for the PR. Great. I am not aware of any issues that this change could induce.

File: crates/shared/src/http_solver.rs [hunk 1]
```diff
@@ -99,8 +99,7 @@ impl HttpSolverApi for DefaultHttpSolverApi {
             .checked_sub(Duration::from_secs(1))
             .ok_or_else(|| anyhow!("no time left to send request"))?;
 
-        let mut url = self.base.clone();
-        url.set_path("/solve");
+        let mut url = self.base.clone().join("solve")?;
```
Comment (nlordell) [inline crates/shared/src/http_solver.rs:102]: nit: `Url::join` already takes a reference and returns a new `Url` instance, so `.clone()` here is superfluous.

```suggestion
        let mut url = self.base.join("solve")?;
```

I tried this out in [the Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=87cdbc6b4ed4ad09bad989780ef18d49) and it seems to work.

**Review by nlordell — APPROVED** [review]
Thanks!

Just one tiny change suggestion, but approving regardless as I don't think its super important.

**Comment by vkgnosis** [comment]
We do have a solver configured with a custom path at the moment so for them the path they observe will change.

**Pull request merged**